cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -259,14 +260,10 @@ protected void rebalanceForBroker(Broker broker,
     }
 
     // Update broker ids over the balance limit for logging purposes.
-    if (requireLessLoad && rebalanceByMovingLoadOut(broker, clusterModel, optimizedGoals,
-                                                    REPLICA_MOVEMENT, excludedTopics)) {
-      _brokerIdsAboveBalanceUpperLimit.add(broker.id());
-      LOG.debug("Failed to balance {} for broker {} with replica and leader movements to reduce load.", resource(), broker.id());
-    } else if (requireMoreLoad && rebalanceByMovingLoadIn(broker, clusterModel, optimizedGoals,
-                                                          REPLICA_MOVEMENT, excludedTopics)) {
-      _brokerIdsUnderBalanceLowerLimit.add(broker.id());
-      LOG.debug("Failed to balance {} for broker {} with replica and leader movements to increase load.", resource(), broker.id());
+    if (requireLessLoad) {
```

Comment:
Is the `else` case of these conditions executed when we actually expect it to be executed?
I am having difficulty in simulating a case where the resource is successfully balanced by moving leaders and replicas.